job_creator.php

```diff
@@ -368,7 +368,21 @@ private function createPhpunitJobs(
     private function getListOfPhpVersionsByBranchName(): array
     {
         $branch = $this->getInstallerBranch();
-        return MetaData::PHP_VERSIONS_FOR_CMS_RELEASES[$branch] ?? MetaData::PHP_VERSIONS_FOR_CMS_RELEASES['4'];
+        if (isset(MetaData::PHP_VERSIONS_FOR_CMS_RELEASES[$branch])) {
+            return MetaData::PHP_VERSIONS_FOR_CMS_RELEASES[$branch];
+        }
+        // Fallback to using a CMS major based on the version of PHP in composer.json
+        $json = $this->getComposerJsonContent();
+        $cmsMajor = '4';
+        if ($json) {
+            $php = $json->require->php ?? null;
+            $cmsMajor = match($php) {
+                '^8.1' => '5',
+                '^8.3' => '6',
+                default => '4',
+            };
```

Comment:
Let's not hardcode things, this is only gonna kick us in the face later.
We have a mapping of `PHP_VERSIONS_FOR_CMS_RELEASES`. We should do something like
```php
foreach (MetaData::PHP_VERSIONS_FOR_CMS_RELEASES as $cmsVersion => $phpVersions) {
    /* Insert code here to ignore non-majors, e.g. "4.9" */
    // Get the first list of PHP versions that includes our version
    if (in_array($php, $phpVersions) {
        return $phpVersions;
    }
}

// Fallback to the PHP versions allowed for the lowest supported major release line
return MetaData::PHP_VERSIONS_FOR_CMS_RELEASES[MetaData::LOWEST_SUPPORTED_CMS_MAJOR];
```

Reply:
Or frankly just return `[$php]` since we know that's a supported version for this repo, I'd be fine with that too.

Reply:
Updated